vpr/src/server/gateio.cpp

```diff
@@ -0,0 +1,291 @@
+            std::optional<int> job_id_opt = comm::TelegramParser::try_extract_field_job_id(message);
+            std::optional<int> cmd_opt = comm::TelegramParser::try_extract_field_cmd(message);
+            std::optional<std::string> options_opt = comm::TelegramParser::try_extract_field_options(message);
+            if (job_id_opt && cmd_opt && options_opt) {
+                TaskPtr task = std::make_unique<Task>(job_id_opt.value(), cmd_opt.value(), options_opt.value());
+                const comm::TelegramHeader& header = telegram_frame->header;
+                m_logger.queue(LogLevel::Info, "received:", header.info(), task->info(/*skipDuration*/true));
+                std::unique_lock<std::mutex> lock(m_tasks_mutex);
+                m_received_tasks.push_back(std::move(task));
+            } else {
+                m_logger.queue(LogLevel::Error, "broken telegram detected, fail extract options from", message);
+            }
+        }
+    }
+
+    return status;
+}
+
+GateIO::ActivityStatus GateIO::handle_client_alive_tracker(sockpp::tcp6_socket& client, std::unique_ptr<ClientAliveTracker>& client_alive_tracker_ptr) {
+    ActivityStatus status = ActivityStatus::WAITING_ACTIVITY;
+    if (client_alive_tracker_ptr) {
+        /// handle sending echo to client
+        if (client_alive_tracker_ptr->is_time_to_sent_echo()) {
+            comm::TelegramHeader echo_header = comm::TelegramHeader::construct_from_data(comm::ECHO_DATA);
+            std::string message = echo_header.buffer().to_string();
+            message.append(comm::ECHO_DATA);
+            try {
+                std::size_t bytes_sent = client.write(message);
+                if (bytes_sent == message.size()) {
+                    m_logger.queue(LogLevel::Detail, "sent", comm::ECHO_DATA);
+                    client_alive_tracker_ptr->on_echo_sent();
+                }
+            } catch(...) {
+                m_logger.queue(LogLevel::Debug, "fail to sent", comm::ECHO_DATA);
+                status = ActivityStatus::COMMUNICATION_PROBLEM;
+            }
+        }
+
+        /// handle client timeout
+        if (client_alive_tracker_ptr->is_client_timeout()) {
+            m_logger.queue(LogLevel::Error, "client didn't respond too long");
+            status = ActivityStatus::COMMUNICATION_PROBLEM;
+        }
+    }
+
+    return status;
+}
+
+void GateIO::handle_activity_status(ActivityStatus status, std::unique_ptr<ClientAliveTracker>& client_alive_tracker_ptr, bool& is_communication_problem_detected) {
+    if (status == ActivityStatus::CLIENT_ACTIVITY) {
+        if (client_alive_tracker_ptr) {
+            client_alive_tracker_ptr->on_client_activity();
+        }
+    } else if (status == ActivityStatus::COMMUNICATION_PROBLEM) {
+        is_communication_problem_detected = true;
+    }
+}
+
+void GateIO::start_listening()
+{
```

Comment:
move "{" to the previous line

Reply:
fixed https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/2526/commits/531f48511c23991db7fc2e3fe2789c62cde80097